homeassistant/components/zwave_mqtt/entity.py

```diff
@@ -0,0 +1,272 @@
+        }
+        # device with multiple instances is split up into virtual devices for each instance
+        if node_instance > 1:
+            parent_dev_id = create_device_id(node)
+            device_info["name"] += f" - Instance {node_instance}"
+            device_info["via_device"] = (DOMAIN, parent_dev_id)
+        return device_info
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        return {const.ATTR_NODE_ID: self.values.primary.node.node_id}
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        node = self.values.primary.node
+        return f"{create_device_name(node)}: {self.values.primary.label}"
+
+    @property
+    def unique_id(self):
+        """Return the unique_id of the entity."""
+        return self.values.values_id
+
+    @property
+    def available(self) -> bool:
+        """Return entity availability."""
+        # Use OZW Daemon status for availability.
+        instance_status = self.values.primary.ozw_instance.get_status()
+        return instance_status and instance_status.status in [
+            "driverAllNodesQueriedSomeDead",
+            "driverAllNodesQueried",
+            "driverAwakeNodesQueried",
+        ]
+
+    @callback
+    def _value_changed(self, value):
+        """Call when a value from ZWaveDeviceEntityValues is changed.
+
+        Should not be overridden by subclasses.
+        """
+        if value.value_id_key in (v.value_id_key for v in self.values if v):
+            self.on_value_update()
+            self.async_write_ha_state()
+
+    @callback
+    def _value_added(self):
+        """Call when a value from ZWaveDeviceEntityValues is added.
+
+        Should not be overridden by subclasses.
+        """
+        self.on_value_update()
+
+    @callback
+    def _instance_updated(self, new_status):
+        """Call when the instance status changes.
+
+        Should not be overridden by subclasses.
+        """
+        self.on_value_update()
+        self.async_write_ha_state()
+
+    async def _delete_callback(self, values_id):
+        """Remove this entity."""
+        if not self.values:
+            return  # race condition: delete already requested
+        if values_id == self.values.values_id:
+            await self.async_remove()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Call when entity will be removed from hass."""
+        # cleanup OZW listeners
+        self.options.listeners[EVENT_VALUE_CHANGED].remove(self._value_changed)
+        self.options.listeners[EVENT_INSTANCE_STATUS_CHANGED].remove(
+            self._instance_updated
+        )
+
+
+def create_device_name(node: OZWNode):
+    """Generate sensible (short) default device name from a OZWNode."""
+    if node.meta_data["Name"]:
+        dev_name = f'{node.meta_data["Name"]}'
+    else:
+        dev_name = f"{node.node_product_name}"
+    return dev_name
```

Comment:
No need for string formatting
```suggestion
    if node.meta_data["Name"]:
        return node.meta_data["Name"]

    return node.node_product_name
```